lib/factory.js

```diff
@@ -0,0 +1,156 @@
+/*!
+ * body-parser
+ * Copyright(c) 2014 Jonathan Ong
+ * Copyright(c) 2014-2015 Douglas Christopher Wilson
+ * MIT Licensed
+ */
+
+'use strict'
+
+/**
+ * Module dependencies.
+ * @private
+ */
+
+var bytes = require('bytes')
+var contentType = require('content-type')
+var createError = require('http-errors')
+var debug = require('debug')('body-parser:generic')
```

Comment:
I suggest passing the `debug` function as a parameter to `createBodyParser()`. This will allows the created middlewares to use their namespaces, such as `body-parser:json`, for debug messages.